payments-core/src/test/java/com/stripe/android/view/CardBrandViewTest.kt

```diff
@@ -0,0 +1,119 @@
+package com.stripe.android.view
+
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import android.widget.LinearLayout
+import androidx.appcompat.app.AppCompatActivity
+import androidx.test.core.app.ActivityScenario
+import androidx.test.core.app.ApplicationProvider
+import com.google.common.truth.Truth.assertThat
+import com.stripe.android.R
+import com.stripe.android.model.CardBrand
+import com.stripe.android.utils.createTestActivityRule
+import com.stripe.android.view.CardFormViewTestActivity.Companion.VIEW_ID
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.Robolectric
+import org.robolectric.RobolectricTestRunner
+import java.util.Locale
+
+@RunWith(RobolectricTestRunner::class)
+internal class CardBrandViewTest {
+    private val context: Context = ApplicationProvider.getApplicationContext()
+
+    @get:Rule
+    val testActivityRule = createTestActivityRule<CardBrandViewTestActivity>(useMaterial = true)
+
+    @Test
+    fun `networkParams should be present when brand is unknown and preferred networks is set`() = runCardBrandViewTest {
+        cardBrandView.apply {
+            isCbcEligible = true
+            possibleBrands = listOf(CardBrand.CartesBancaires, CardBrand.MasterCard)
+            merchantPreferredNetworks = listOf(CardBrand.CartesBancaires)
+        }
+
+        assertThat(cardBrandView.createNetworksParam()?.preferred).isEqualTo(CardBrand.CartesBancaires.code)
+        assertThat(cardBrandView.merchantPreferredNetworkParams()?.preferred).isEqualTo(CardBrand.CartesBancaires.code)
+    }
+
+    @Test
+    fun `networkParams should be present when brand is known, possible cards empty and preferred networks is set`() =
```

Comment:
This should cover the [scenario](https://github.com/stripe/stripe-android/pull/9236/files#r1757205612) you mentioned @amk-stripe